pgcli/main.py

```diff
@@ -984,13 +984,24 @@ def format_arrays(data, headers, **_):
 
     if cur:
         headers = [case_function(utf8tounicode(x)) for x in headers]
-        rows = list(cur)
-        formatted = formatter.format_output(rows, headers, **output_kwargs)
+        if max_width is not None:
+            cur = list(cur)
+        column_types = None
+        if hasattr(cur, 'description'):
+            column_types = []
+            for d in cur.description:
+                if d[1] == 23:
+                    column_types.append(int)
+                elif d[1] == 1700:
+                    column_types.append(float)
+                else:
+                    column_types.append(type(""))
```

Comment:
The variable `text_type` is available in pgcli's `main.py`, so you might try using that instead.